selfdrive/ui/qt/onroad.cc

```diff
@@ -197,7 +200,9 @@ void ExperimentalButton::updateState(const UIState &s) {
   setVisible(cs.getEngageable() || cs.getEnabled());
 
   // button is "checked" if experimental mode is enabled
-  setChecked(sm["controlsState"].getControlsState().getExperimentalMode());
+  if (isEnabled()) {
+    setChecked(sm["controlsState"].getControlsState().getExperimentalMode());
+  }
```

Comment:
What if we always update from controlsState and disallow clicking to change the checked state? Then we make no assumptions about how long the round trip from setting param to controlsd reading it, to it sending it out with cereal?

Reply:
In my opinion, this experimental button may not belong here at all. All other icons in the HUD cannot be clicked, only this one can, and there is no visual difference. This is a very strange thing, and users who are not familiar with it would not understand its purpose. Perhaps such a button should only exist in the settings interface.